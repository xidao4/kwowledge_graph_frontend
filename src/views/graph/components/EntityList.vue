<template>
  <div class="addEntityForm">
    <h3 style="margin-left:40%;padding-top:6%">添加实体</h3>
    <a-divider />
    <a-form
            :form="addForm"
            :label-col="{ lg: {span: 10}, xl: {span: 9} }"
            :wrapper-col="{ lg: {span: 14}, xl: {span: 15} }"
            style="margin-right:16%"
    >
      <a-form-item
              label="实体值："
              :validate-status="name.validateStatus"
              :help="name.errorMsg || nameTips"
              
      >
        <a-input
                v-model="name.value"
                :maxLength="10"
                @change="handleNameChange"
                
        />
      </a-form-item>
      <a-form-item
              label="实体类型："
              :validate-status="type.validateStatus"
              :help="type.errorMsg || typeTips"
      >
        <a-input
                v-model="type.value"
                :maxLength="10"
                @change="handleTypeChange"
        />
      </a-form-item>
    </a-form>
    <a-tag
            style="cursor: pointer; border-style: solid;"
            @click="handleAdd"
            class="tag-item"
            id="addEntityButton"
    >
      <a-icon type="plus" /> 确认增加
    </a-tag>
  </div>
</template>

<script>
import { mapGetters, mapMutations, mapActions } from "vuex";
// import ColorPicker2 from './ColorPicker2';
import { validateEntityName,validateEntityType } from "@/utils/validator.js";


export default {
  name: "EntityList",
  data() {
    return {
      addForm: this.$form.createForm(this, { name: "addEntityForm" }),
      name: {
        value: "",
      },
      type:{
          value: "",
      },
      nameTips: "请输入1-10位实体值，与现有实体值不相同",
      typeTips: "请输入1-10位实体类型",
    };
  },
  props: ["triggerGraphDraw"],
  computed: {
    ...mapGetters([
        "colorList",
        "showGraphNodes",
        "picId",
        "addEntityVisible",
        "currentGraph",
        "currentGraphData",
        "nodeId",
        "currentShowGraphData",
        "nodesTypes",
        'checkList',
        'boardTypes',
    ]),
  },
  methods: {
    ...mapMutations(["add_showGraphNodes","set_addEntityVisible","set_nodeId",'set_currentShowGraphData',
      'set_currentShowGraphData_typesetting',
      'set_currentGraphData','set_currentShowBoard']),
    ...mapActions(["addEntity"]),
    handleAdd() {
      this.name = {
        ...validateEntityName(this.name.value, null, this.showGraphNodes),
        value: this.name.value,
      };
      this.type = {
        ...validateEntityType(this.type.value),
        value: this.type.value,
      };



      if (this.name.errorMsg === null && this.type.errorMsg === null) {

        let model = {
          id:  `node-${this.nodeId}`,
          label: this.name.value,
          oriLabel: this.name.value,
          address: 'cq',
          x: 200+Math.random()*100,
          y: 150+Math.random()*100,
          style:{
            fill:'#E65D6E'
          }
        };
        console.log('myModel',model)
        this.currentGraph.addItem('node',model)
        this.currentGraph.refresh()
        this.currentGraphData.nodes.push({
          id:  `node-${this.nodeId}`,
          class: this.type.value,
          label: this.name.value,
          oriLabel: this.name.value,
        })
        let temp=this.currentGraph.save()
        this.set_currentShowGraphData_typesetting(temp)
        this.set_currentGraphData(temp)
        console.log('添加完后的Data',this.currentGraphData)
        this.set_nodeId();
        console.log('after update',this.currentGraphData)
        this.set_addEntityVisible(false)
        //类型筛选相关
        if(this.nodesTypes.indexOf(this.type.value)===-1){
          this.nodesTypes.push(this.type.value);
        }
        console.log('4444 新增节点有新类型 this.nodesTypes',this.nodesTypes);
        this.checkList.push(true);

        this.name.value = "";
        this.type.value= "";
        this.set_currentShowBoard(this.boardTypes.none)
      }
    },
    handleNameChange() {
      this.name = {
        ...validateEntityName(this.name.value, null, this.showGraphNodes),
        value: this.name.value,
      };
    },
    handleTypeChange() {
      this.type = {
        ...validateEntityType(this.type.value),
        value: this.type.value,
      };
    },
    onClose() {
      this.set_addEntityVisible(false)
    },
  },
  components: {
    // ColorPicker2
  },
};
</script>

<style scoped>
@media screen and (max-width: 992px) {
    .ant-row{
        margin-left: 12%;
    }
}
.tag-item {
  line-height: 28px;
  cursor: pointer;
  margin-left: 40%;
  margin-top: 160px;
}

</style>
