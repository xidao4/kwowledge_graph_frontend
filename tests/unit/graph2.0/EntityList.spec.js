import { mount, createLocalVue } from '@vue/test-utils'
import EntityList from '@/views/graph/components/EntityList.vue'
import Antd from 'ant-design-vue';
import Vuex from 'vuex';

const localVue = createLocalVue();
localVue.use(Antd);
localVue.use(Vuex);

describe('EntityList',()=>{
  let mutations;
  let getters;
  let store;
  let wrapper;
  beforeEach(()=>{
    mutations = {
      set_nodeId:jest.fn(),
      set_currentShowGraphData:jest.fn(),
      set_currentShowGraphData_typesetting:jest.fn(),
      set_currentGraphData:jest.fn(),
      set_currentShowBoard:jest.fn(),
    };
    getters={
      colorList:()=>[],
      picId:()=>'6075ab0c1f3a46144cf5c4c1',
      currentGraph: () => {
        return {
          data: jest.fn(),
          updateLayout: jest.fn()
        }
      },
      currentGraphData:()=>{
        return{
          data: jest.fn()
        }
      },
      nodeId:()=>0,
      currentShowGraphData: () => {
        return {
          typesetting: {},
          force: {}
        }
      },
      boardTypes:()=>{
        return{
          none: '0',
          pie: '1',
          entityList: '2',
          relationList: '3',
          entityEdit: '4',
          relationEdit: '5',
        }
      }
    }
    store = new Vuex.Store({
      state: {},
      mutations,
      getters,
    });
    wrapper = mount(EntityList, {
      store,
      localVue
    });
  })
  it('entityList snapshot', () => {
    expect(wrapper).toMatchSnapshot()
  });
  it("has label input",()=>{
    expect(wrapper.html()).toContain('<label title="实体值：" class="">实体值</label>')
  })
  it("has labelValue input",()=>{
    expect(wrapper.html()).toContain('<label title="实体类型：" class="">实体类型</label>')
  })
  it("has button",()=>{
    expect(wrapper.html()).toContain('确认增加')
  })
  it("input value",async ()=>{
    await wrapper.find('input.ant-input').trigger('keydown',{
      key:'a'
    })
  })
})