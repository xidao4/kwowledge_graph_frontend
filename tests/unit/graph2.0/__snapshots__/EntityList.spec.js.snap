// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EntityList entityList snapshot 1`] = `
<div class="addEntityForm">
  <h3 style="margin-left: 40%; padding-top: 6%;">添加实体</h3>
  <div role="separator" class="ant-divider ant-divider-horizontal"></div>
  <form class="ant-form ant-form-horizontal" style="margin-right: 16%;">
    <div class="ant-row ant-form-item ant-form-item-with-help">
      <div class="ant-col ant-col-lg-10 ant-col-xl-9 ant-form-item-label"><label title="实体值：" class="">实体值</label></div>
      <div class="ant-col ant-col-lg-14 ant-col-xl-15 ant-form-item-control-wrapper">
        <div class="ant-form-item-control"><span class="ant-form-item-children"><input type="text" maxlength="10" class="ant-input"></span>
          <transition-stub appear="true">
            <div class="ant-form-explain">请输入1-10位实体值，与现有实体值不相同</div>
          </transition-stub>
        </div>
      </div>
    </div>
    <div class="ant-row ant-form-item ant-form-item-with-help">
      <div class="ant-col ant-col-lg-10 ant-col-xl-9 ant-form-item-label"><label title="实体类型：" class="">实体类型</label></div>
      <div class="ant-col ant-col-lg-14 ant-col-xl-15 ant-form-item-control-wrapper">
        <div class="ant-form-item-control"><span class="ant-form-item-children"><input type="text" maxlength="10" class="ant-input"></span>
          <transition-stub appear="true">
            <div class="ant-form-explain">请输入1-10位实体类型</div>
          </transition-stub>
        </div>
      </div>
    </div>
  </form>
  <transition-stub id="addEntityButton" class="tag-item" style="cursor: pointer; border-style: solid;"><span class="ant-tag"><i aria-label="icon: plus" class="anticon anticon-plus"><svg viewBox="64 64 896 896" data-icon="plus" width="1em" height="1em" fill="currentColor" aria-hidden="true" focusable="false" class=""><path d="M482 152h60q8 0 8 8v704q0 8-8 8h-60q-8 0-8-8V160q0-8 8-8z"></path><path d="M176 474h672q8 0 8 8v60q0 8-8 8H176q-8 0-8-8v-60q0-8 8-8z"></path></svg></i> 确认增加
  </span></transition-stub>
</div>
`;
